### DISCUSSION TITLE
Saving Multiple rows at once

### DISCUSSION THREAD

**Discussion by insomniacj93** [Budibase/budibase#8278] [Help]
Hello, I've been using the Continue if/ Stop if solution to save multiple rows at once.
The issue I'm facing now is that with this kind of solution I can't add a Navigate to or Update field value (to disable the form for e.g.), leaving the user at the same form, able to save again (creating double entries in db) or even change the inputs and create an even greater mess.

The necessity behind this solution is that I need to register each quantity with a different id (e.g. id301 : 45). I rely on up to 30 different form fields for each (id#1, id#2, ... / quantity#1, quantity#2, ...) and check if the very next one is = to the default value ( 0 ). I ended up with 62 actions on a single press of a button (no problem there as budibase seems to take it like a champ!) But besides seeming a little bit rudimentary, leaving the user just there after everything is saved isn't a good practice at all.

I tried using automations too, with automations I could simply save the 30 rows (saving both values to db) and a Navigate to after it, but that means that budibase will be applying the trigger 30x when we only used 10 for example, the 20 extra actions had no purpose and are registered in the loggs.

![print24](https://user-images.githubusercontent.com/107686004/196118815-e67a1588-3887-4b38-b29b-fc00234d17d3.png)

I just wish there was a way to restrict user input after the rows are saved.
Thank you in advance!

**Comment by melohagan** [comment]
Hey @insomniacj93 

62 actions - wow! 

I don't think I'm fully grasping the use case here. Would you mind sharing your app export to support@budibase.com ? And/or maybe some extra screenshots of your form and the state you expect would help to!

**Reply by insomniacj93** [reply]
@melohagan Sorry for the late reply!!

I solved the issue by adding a checkbox to the form which is set to 1 everytime the confirm button is pressed. With this everytime the user pressed said confirmation button the checkbox is set to 1 and runs all teh actions below until its done. The form fields will be hidden when said checkbox is set to 1 so the users can no longer alter data, they end up at the same page but without capabilities to alter anything. So yeah "solved" ^^'

Edit: Also the confirm button is hidden so the user wont keep adding rows.
